lib/ansible/parsing/vault/__init__.py

```diff
@@ -625,9 +626,6 @@ def encrypt(self, plaintext, secret=None, vault_id=None):
         if is_encrypted(b_plaintext):
             raise AnsibleError("input is already encrypted")
 
-        if not self.cipher_name or self.cipher_name not in CIPHER_WRITE_WHITELIST:
-            self.cipher_name = u"AES256"
```

Comment:
We do want to keep the CIPHER_WRITE_WHITELIST check here for sanity.  It could be changed to throw an error, though, as cipher_name should be set in __init__ with your change.

Reply:
Re added the if and changed to throw an exception